db/query_test.go

```diff
@@ -568,3 +569,128 @@ func countQueryResults(results sgbucket.QueryResultIterator) int {
 	}
 	return count
 }
+
+func TestQueryChannelsActiveOnlyWithLimit(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test require Couchbase Server")
+	}
+
+	db, testBucket := setupTestDB(t)
+	defer testBucket.Close()
+	defer tearDownTestDB(t, db)
+
+	var sequence = 1
+	seqLogEntries := make(map[int]LogEntry)
+	body := Body{"channels": []string{"ABC"}}
+
+	createDocs := func(name string, revs int) (startSeq, endSeq uint64) {
+		for i := 1; i <= revs; i++ {
+			body["version"] = "1-a"
+			id := name + strconv.Itoa(i)
+			doc, revId, err := db.PutExistingRevWithBody(id, body, []string{"1-a"}, false)
+			require.NoError(t, err, "Couldn't create document")
+			require.Equal(t, "1-a", revId)
+			seqLogEntries[sequence] = LogEntry{DocID: doc.ID, RevID: revId}
+			sequence++
+			if i == 1 {
+				startSeq = doc.Sequence
+			}
+			endSeq = doc.Sequence
+		}
+		delete(body, "version")
+		return startSeq, endSeq
+	}
+
+	deleteDocs := func(name string, revs int) (startSeq, endSeq uint64) {
+		for i := 1; i <= revs; i++ {
+			id := name + strconv.Itoa(i)
+			body[BodyDeleted] = true
+			doc, revId, err := db.PutExistingRevWithBody(id, body, []string{"2-a", "1-a"}, false)
+			require.NoError(t, err, "Couldn't create document")
+			require.Equal(t, "2-a", revId, "Couldn't create tombstone revision")
+			seqLogEntries[sequence] = LogEntry{DocID: doc.ID, RevID: revId, Flags: uint8(0x1)}
+			sequence++
+			if i == 1 {
+				startSeq = doc.Sequence
+			}
+			endSeq = doc.Sequence
+		}
+		return startSeq, endSeq
+	}
+
+	checkDocs := func(entries LogEntries, sequence int) {
+		for _, entry := range entries {
+			assert.Equal(t, seqLogEntries[sequence].DocID, entry.DocID)
+			assert.Equal(t, seqLogEntries[sequence].RevID, entry.RevID)
+			assert.Equal(t, seqLogEntries[sequence].Flags, entry.Flags)
+			sequence++
+		}
+	}
+
+	// Create 10 active documents and 2 inactive documents.
+	startSeq, endSeq := createDocs("alice", 10)
+	_, endSeq = deleteDocs("alice", 2)
```

Comment:
For this test, we want to create documents in the following states, to validate the query's interaction with the bitflag.
- flags=0. (createDocs)
- flags = deleted (deleteDocs)
- flags = branched  (two branches, one branch is tombstoned)
- flags = branched|deleted (two branches, both branches tombstoned)
- flags = branched|conflict (two branched, neither branch tombstoned)

It would be ideal to create n docs for each of these states, and use those to validate query responses.